nav2-usability/scenario-code/dynamic-linking/dynamic_linking_beamer.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:beamer/beamer.dart';
+
+void main() {
+  runApp(WishlistApp());
+}
+
+class Wishlist {
+  final String id;
+
+  Wishlist(this.id);
+}
+
+class AppState extends ChangeNotifier {
+  final List<Wishlist> wishlists = <Wishlist>[];
+
+  Wishlist? _selectedWishlist;
+
+  Wishlist? get selectedWishlist => _selectedWishlist;
+
+  set selectedWishlist(Wishlist? w) {
+    _selectedWishlist = w;
+    notifyListeners();
+  }
+
+  void addWishlist(Wishlist wishlist) {
+    wishlists.add(wishlist);
+    notifyListeners();
+  }
+}
+
+class WishlistLocation extends BeamLocation {
+  WishlistLocation(AppState appState, BeamState state)
+      : _appState = appState,
+        super(state) {
+    _appState.addListener(
+      () => update(
+        (state) => (state.copyWith(
+          pathBlueprintSegments: _appState.selectedWishlist != null
+              ? ['wishlist', ':wishlistId']
+              : ['wishlist'],
+          pathParameters: _appState.selectedWishlist != null
+              ? {'wishlistId': _appState.selectedWishlist!.id}
+              : {},
+        )),
+      ),
+    );
```

Comment:
This is probably easier to read:

```suggestion
    _appState.addListener(
      () => update((state) {
        var selectedWishlist = _appState.selectedWishlist;
        if (selectedWishlist != null) {
          return state.copyWith(
              pathBlueprintSegments: ['wishlist', ':wishlistId'],
              pathParameters: {'wishlistId': selectedWishlist.id});
        }
        return state
            .copyWith(pathBlueprintSegments: ['wishlist'], pathParameters: {});
      }),
    );

```